lib/vm.js

```diff
@@ -83,19 +78,22 @@ exports.isContext = binding.isContext;
 
 // Remove all SIGINT listeners and re-attach them after the wrapped function
 // has executed, so that caught SIGINT are handled by the listeners again.
-function sigintHandlersWrap(fn) {
+function sigintHandlersWrap(fn, thisArg, argsArray) {
   // Using the internal list here to make sure `.once()` wrappers are used,
   // not the original ones.
   let sigintListeners = process._events.SIGINT;
-  if (!Array.isArray(sigintListeners))
-    sigintListeners = sigintListeners ? [sigintListeners] : [];
-  else
+  if (!sigintListeners)
```

Comment:
What about moving this to the conditional in the caller function (`options && options.breakOnSigint`) to avoid creating the `argsArray` and calling `fn.apply()`?